src/Core/Services/CryptoService.cs

```diff
@@ -389,28 +389,27 @@ public async Task ToggleKeyAsync()
             await SetKeyAsync(key);
         }
 
-        public async Task<SymmetricCryptoKey> MakeKeyAsync(string password, string salt,
-            KdfType? kdf, int? kdfIterations)
+        public async Task<SymmetricCryptoKey> MakeKeyAsync(string password, string salt, KdfConfig kdfConfig)
         {
             byte[] key = null;
-            if (kdf == null || kdf == KdfType.PBKDF2_SHA256)
+            if (kdfConfig.Type == null || kdfConfig.Type == KdfType.PBKDF2_SHA256)
             {
-                if (kdfIterations == null)
+                if (kdfConfig.Iterations == null)
                 {
-                    kdfIterations = 5000;
+                    kdfConfig.Iterations = 5000;
                 }
-                if (kdfIterations < 5000)
+                if (kdfConfig.Iterations < 5000)
                 {
                     throw new Exception("PBKDF2 iteration minimum is 5000.");
                 }
                 key = await _cryptoFunctionService.Pbkdf2Async(password, salt,
-                    CryptoHashAlgorithm.Sha256, kdfIterations.Value);
+                    CryptoHashAlgorithm.Sha256, kdfConfig.Iterations.Value);
             }
-            else if (kdf == KdfType.Argon2id)
+            else if (kdfConfig.Type == KdfType.Argon2id)
             {
-                var iterations = kdfIterations.Value;
-                const int parallelism = 1;
-                const int memory = 1024 * 16; // 16 MiB
+                var iterations = kdfConfig.Iterations.GetValueOrDefault(Constants.Argon2Iterations);
```

Comment:
Didn't you add some validation checks here on clients repo?

Reply:
Ah, yeah good catch. That's still missing.

Reply:
Added checks in afd42406060d62cb0726e52af49a9df6ca5deeea